src/adapter/http/src/smart_protocol/protocol_dataset_helper.rs

```diff
@@ -220,15 +220,12 @@ pub async fn dataset_append_metadata(
             // Collect only the latest upstream dataset IDs
             new_upstream_ids.clear();
             for new_input in transform.inputs.iter() {
-                if let Some(id) = &new_input.id {
+                if let Some(id) = new_input.dataset_ref.id() {
                     new_upstream_ids.push(id.clone());
                 } else {
-                    return Err(AppendError::InvalidBlock(
-                        AppendValidationError::InvalidEvent(InvalidEventError::new(
-                            block.event,
-                            "Transform input with unresolved ID",
-                        )),
-                    ));
+                    // Input references must be resolved to IDs here, but we
+                    // ignore the errors and let the metadata chain reject this
+                    // event
```

Comment:
Previously we had:
```rust
                    return Err(AppendError::InvalidBlock(
                        AppendValidationError::InvalidEvent(InvalidEventError::new(
                            block.event,
                            "Transform input with unresolved ID",
                        )),
                    ));
```

Q: Could you please clarify: how will the event be rejected?

Reply:
See `metadata_chain_impl.rs` - it contains a bunch of validation logic and is our "ground source of truth" on what is considered a valid metadata.

Ideally we want to make states like this unrepresentable in the type system and I already have a ticket for that:
https://github.com/open-data-fabric/open-data-fabric/issues/73